nemo_skills/pipeline/prepare_data.py

```diff
@@ -0,0 +1,127 @@
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+from typing import List
+
+import typer
+
+from nemo_skills.pipeline.app import app, typer_unpacker
+from nemo_skills.pipeline.run_cmd import run_cmd as _run_cmd
+from nemo_skills.pipeline.utils import get_cluster_config, get_env_variables
+from nemo_skills.utils import get_logger_name, setup_logging
+
+LOG = logging.getLogger(get_logger_name(__file__))
+
+
+@app.command(context_settings={"allow_extra_args": True, "ignore_unknown_options": True})
+@typer_unpacker
+def prepare_data(
+    ctx: typer.Context,
+    cluster: str = typer.Option(
+        None,
+        help="One of the configs inside config_dir or NEMO_SKILLS_CONFIG_DIR or ./cluster_configs.",
+    ),
+    data_dir: str = typer.Option(
+        None,
+        help="Path to the data directory. If not specified, will use the default nemo_skills/dataset path. "
+        "If a custom path is provided, you'd need to set it as NEMO_SKILLS_DATA_DIR environment variable "
+        "or pass it as an argument to the `ns eval` command.",
+    ),
+    expname: str = typer.Option("prepare-data", help="Experiment name for data preparation"),
+    partition: str = typer.Option(None, help="Slurm partition to use"),
+    time_min: str = typer.Option(None, help="Time-min slurm parameter"),
+    num_gpus: int | None = typer.Option(None, help="Number of GPUs to use"),
+    num_nodes: int = typer.Option(1, help="Number of nodes to use"),
+    mount_paths: str = typer.Option(None, help="Comma separated list of paths to mount"),
+    run_after: List[str] = typer.Option(None, help="List of expnames that this job depends on before starting"),
+    reuse_code: bool = typer.Option(True, help="Whether to reuse code from previous experiments"),
+    reuse_code_exp: str = typer.Option(None, help="Experiment to reuse code from"),
+    config_dir: str = typer.Option(None, help="Custom cluster config directory"),
+    with_sandbox: bool = typer.Option(False, help="Start a sandbox container alongside"),
+    log_dir: str = typer.Option(None, help="Custom location for slurm logs"),
+    exclusive: bool = typer.Option(True, "--not_exclusive", help="Disable --exclusive flag"),
+    check_mounted_paths: bool = typer.Option(False, help="Check mounted paths availability"),
+):
+    """Prepare datasets by running python -m nemo_skills.dataset.prepare.
+
+    Run `python -m nemo_skills.dataset.prepare --help` to see other supported arguments.
+    """
+    setup_logging(disable_hydra_logs=False, use_rich=True)
+    extra_arguments = f'{" ".join(ctx.args)}'
+    command = f"python -m nemo_skills.dataset.prepare {extra_arguments}"
```

Comment:
Hmm the dataset preparation command should be taken as input in typer (with this as default). That way a user can point to their own script to process the dataset.. Can be done later though.

Reply:
Also if its generating the dataset locally and then copying onto cluster .. then you need to delete the copy in /nemo_run/code/ns/dataset no? Or two copies will remain, one in data dir and one inside the nemo-run folder.

Reply:
> Hmm the dataset preparation command should be taken as input in typer (with this as default).
Let's do that in another PR

Reply:
> then you need to delete the copy in /nemo_run/code/ns/dataset no
There will be two copies, that's true. I thought about using mv instead of cp, but a little hesitant to do that as the dataset folder in code might contain files that we didn't create and so I don't want to be removing them. I agree that probably 99% of the cases that's safe to do, but I imagine there might be some edge-cases where it's not what the user expects (e.g. if reusing code across multiple experiments). And in any case, people need to be cleaning up the nemo-run folder from time to time. 

I'd suggest we go with copy behavior for now and if this becomes an issue we can address this later